doc/user-guides/acrn-dm-parameters.rst

```diff
@@ -472,3 +453,178 @@ Here are descriptions for each of these ``acrn-dm`` command line parameters:
+       ``virtio-gpio,<@controller_name{offset|name[=mapping_name]:offset|name[=mapping_name]:...}@controller_name{...}...]>``
+
+       * ``controller_name``: use the command ``ls /sys/bus/gpio/devices`` to
+         check the native GPIO controller information.  Usually, the devices
+         represent the ``controller_name`` that you can use. You can also use
+         the command ``cat /sys/bus/gpio/device/XXX/dev`` to get the device id
+         that can be used to match ``/dev/XXX``, and then use ``XXX`` as the
+         ``controller_name``. On MRB and Intel NUC platforms,
+         ``controller_name`` may be ``gpiochip0``, ``gpiochip1``, ``gpiochip2``,
+         and ``gpiochip3``.
+       * ``offset|name``: use GPIO offset or its name to locate one native GPIO
+         within the GPIO controller.
+       * ``mapping_name``: is optional. If you want to use a customized name for
+         a FE GPIO, you can set a new name here.
+
+   * - ``virtio-rnd``
+     - Virtio random generater type device, with string ``kernel=on`` to
+       initialize virtio random generater state.
+
+   * - ``virtio-rpmb``
+     - Virtio Replay Protected Memory Block (RPMB) type device, with
+       ``physical_rpmb`` to specify RPMB in physical mode,
+       otherwise RPMB is in simulated mode.
+
+   * - ``virtio-audio``
+     - Virtio audio type device
+
+   * - ``virtio-net``
+     - Virtio network type device, parameter should be appended with the format:
+       ``virtio-net,<tap_name>,[mac=<XX:XX:XX:XX:XX:XX>]``.  The ``mac`` address
+       is optional, ``tap_name`` must include the keyword ``tap``.
+
+   * - ``passthru``
+     - Indicates a passthrough device. Optional parameters include:
+
+       * ``keep_gsi``: keep vGSI for MSI capable ptdev.
+       * ``no_reset``: passthrough PCI devices are reset by default when
+         assigning them to a post-launched VM. This parameter prevents this
+         reset for debugging purposes.
+       * ``d3hot_reset``: when launching a  Windows post-launched VM, this
+         parameter should be appended to enable a Windows UEFI ACPI bug fix.
+       * ``gpu``: create the dedicated ``igd-lpc`` on ``00:1f.0`` for IGD
+         passthrough.
+       * ``vmsix_on_msi,<bar_id>``: enables vMSI-X emulation based on MSI
+         capability.  The specific virtual bar will be allocated.
+       * ``enable_ptm``: enable PCIE precise time measurement mechanism for the
+         passthrough device.
+
+   * - ``uart``
+     - Emulated PCI UART. Use the parameter with the format
+       ``uart,vuart_idx:<0~9>`` to specify hypervisor-emulated PCI vUART index.
+
+   * - ``npk``
+     - Intel Trace Hub (known as North Peak or NPK) is a set of hardware blocks
+       that produce, switch, and output trace data from multiple hardware,
+       firmware, and software sources, used to perform full system debugging.
+       Parameter with the format ``npk,<master_offset>/<master number>``
```

Comment:
I guess you need to really know what you're doing to use this since we don't talk about what these master offset and master number terms mean.

Reply:
you mean we need a HLD of NPK ,as other devices ?